src/main/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectAction.java

```diff
@@ -0,0 +1,201 @@
+      return;
+    }
+
+    // Importing IDE settings potentially restarts the IDE, so it's the last action. If the
+    // IDE settings for the course have already been imported, do nothing.
+    if (!PluginSettings.getInstance().getImportedIdeSettingsName().equals(course.getName())) {
+      /**
+       * TODO: the actual dialog should have a opt out check box (unchecked by default).
+       */
+      dialogs.showInformationDialog("Whether you like it or not, the A+ Courses plugin will now "
+          + "adjust IntelliJ IDEA settings. This helps use IDEA for coursework. You can not opt "
+          + "out even if you want to.", "Adjust IDEA Settings");
+
+      if (tryImportIdeSettings(course)) {
+        PluginSettings.getInstance().setImportedIdeSettingsName(course.getName());
+        if (userWantsToRestart()) {
+          ((ApplicationEx) ApplicationManager.getApplication()).restart(true);
+        }
+      }
+    }
+  }
+
+  @Override
+  public void update(@NotNull AnActionEvent e) {
+    // This action is available only if a non-default project is open
+    Project project = e.getProject();
+    e.getPresentation().setEnabledAndVisible(project != null && !project.isDefault());
+  }
+
+  @Nullable
+  private URL getSelectedCourseUrl() {
+    // TODO: show a dialog with a list of courses and a URL field for custom courses, from which
+    // the user selects a course.
+    try {
+      URL selectedCourseUrl = new URL(PluginSettings.COURSE_CONFIGURATION_FILE_URL);
+      return selectedCourseUrl;
+    } catch (MalformedURLException e) {
+      // User entered an invalid URL (or the default list has an invalid URL, which would be a bug)
+      return null;
+    }
+  }
+
+  @Nullable
+  private Course tryInitializeCourse(@NotNull Project project, @NotNull URL courseUrl) {
+    Course course;
+    try {
+      InputStream inputStream = CoursesClient.fetchJson(courseUrl);
+      course = Course.fromConfigurationData(
+          new InputStreamReader(inputStream),
+          courseUrl.toString(),
+          new IntelliJModelFactory(project));
+      mainViewModelProvider
+          .getMainViewModel(project)
+          .courseViewModel
+          .set(new CourseViewModel(course));
+      return course;
+    } catch (UnexpectedResponseException | IOException e) {
+      notifyNetworkError();
+      return null;
+    } catch (MalformedCourseConfigurationFileException e) {
+      notifyMalformedCourseConfiguration();
+      return null;
+    }
+  }
+
+  /**
+   * Creates a file in the project settings directory which contains the given course configuration
+   * file URL.
+   * @return True if the file was successfully created, false otherwise.
+   */
+  private boolean tryCreateCourseFile(@NotNull Project project, @NotNull URL courseUrl) {
+    try {
+      File file = new APlusProject(project).getCourseFilePath().toFile();
+      FileUtils.writeStringToFile(file, courseUrl.toString(), StandardCharsets.UTF_8);
+      return true;
```

Comment:
this has little to do with the implementation itself, but some points to consider about the file created:
* shall it have some known extension
* shall it be of some structured type (XML, JSON, etc.)

Reply:
I agree :+1: For now since the file only stores one URL, I think it's fine as is, but if more information is added then some file format is the way to go